### PR TITLE
[#109] Created Page template story in Storybook.

### PR DESCRIPTION
**Pull request by joshua-salsadigital**
Closes https://github.com/civictheme/uikit/issues/109

## Checklist before requesting a review

- [x] I have formatted the subject to include the issue number
  as `[#123] Verb in past tense with a period at the end.`
- [x] I have provided information in the `Changed` section about WHY something was
  done if this was a bespoke implementation.
- [ ] I have performed a self-review of my code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have run new and existing relevant tests locally with my changes,
  and they have passed.
- [ ] I have provided screenshots, where applicable.

## Changed

1. Created 2 Storybook stories for page template using examples from Figma.

## Screenshots
Screenshot 2024-04-10 at 4 28 15 PM

### CODE REVIEW

**Review by AlexSkrypnyk — CHANGES_REQUESTED** [review]
@joshua-salsadigital 
This looks great.

however, the exposed controls need to be done a bit different: we want to expose only Organism-level controls with a minimum of options - for now, just the `Theme`. Please remove other controls.

The Knobs panel should have `General` + Tabs for each Organism used.

**Review by AlexSkrypnyk — CHANGES_REQUESTED** [review]
@joshua-salsadigital 
Thank you for putting through the updates.
I've reviewed it and the Storybook itself is extremely slow.

Can we please remove all of the props except for `Theme`: the theme should be passed to all component's props to have all of the changed. The rest of component's props should default to the defaults found in their stories.

**Comment by joshua-salsadigital** [comment]
> @joshua-salsadigital Thank you for putting through the updates. I've reviewed it and the Storybook itself is extremely slow.
> 
> Can we please remove all of the props except for `Theme`: the theme should be passed to all component's props to have all of the changed. The rest of component's props should default to the defaults found in their stories.

Updated code, to pass default values for components, and pass theme value to components.

File: components/04-templates/page/page.stories.js [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    generalKnobTab,
+  );
+
+  const generalKnobs = {
+    theme,
+    header_theme: theme,
+    footer_theme: theme,
+  };
+
+  generalKnobs.logo = Logo('Logo', false, false, theme);
```
Comment (joshua-salsadigital) [inline components/04-templates/page/page.stories.js:52]: Refactored possible component's stories to allow a reuse of their stories without the need to expose all those additional controls.

File: components/04-templates/page/page.stories.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      'content_top',
+      'content_bottom',
+    ]),
+  }).join(' ');
+
+  generalKnobs.slides = slides;
+
+  generalKnobs.list_promo = List('List', false, theme);
+
+  generalKnobs.list_navigation = List('List', false, theme, 'navigation-card');
```
Comment (joshua-salsadigital) [inline components/04-templates/page/page.stories.js:75]: Note, since now we are using the default. it will be listing with filters pagination.

**Review by AlexSkrypnyk — CHANGES_REQUESTED** [review]
@joshua-salsadigital 
Can we please replace all these added parameters with a `defaults` object. Then, the caller can simply pass the desired defaults.

**Comment by AlexSkrypnyk** [comment]
Closed in favour of https://github.com/civictheme/uikit/pull/191

**Pull request closed without merging**